dbldatagen/data_generator.py

```diff
@@ -1604,3 +1617,51 @@ def scriptMerge(self, tgtName=None, srcName=None, updateExpr=None, delExpr=None,
             result = HtmlUtils.formatCodeAsHtml(results)
 
         return result
+
+    @staticmethod
+    def fromDict(options):
+        """ Creates a data generator from a dictionary of options.
+
+            :param options: Dictionary with data generator options (e.g. "name", "rows")
+            :return: A data generator with the specified options
+        """
+        return DataGenerator(**options)
+
+    @staticmethod
+    def fromFile(path):
+        """ Creates a data generator from options loaded from a JSON or YAML file.
+
+            :param path: File path to a JSON or YAML file containing data generation options
+            :return: A data generator with the specified options
+        """
+        if path.endswith("yml") or path.endswith("yaml"):
+            return DataGenerator.fromYaml(path)
+        if path.endswith("json"):
+            return DataGenerator.fromJson(path)
+        raise ValueError("File type must be '.json' or '.yml'")
+
+    @staticmethod
+    def fromJson(path):
```

Comment:
Rather than taking a path, pass a string containing the definition to method

Calling code should be responsible for loading string 

it could be from dbfs, from a database, from unity catalog